src/main/java/org/iot/dsa/dslink/dynamodb/Util.java

```diff
@@ -0,0 +1,619 @@
+                        break;
+                }
+            } else {
+                System.out.println(" Wrong format for " + key);
+            }
+        }
+        return valueMap;
+    }
+
+    public static List getList(DSList list){
+        List listA = new ArrayList();
+        for (DSElement en : list) {
+            if(en.getElementType().equals(DSValueType.NUMBER)){
+                listA.add(Double.valueOf(en.toDouble()));
+            }else {
+                listA.add(en.toString());
+            }
+        }
+        return listA;
+    }
+
+    private static Map<String, AttributeValue> getAttributeKey(DSMap keyMap){
+        Map<String, AttributeValue> startKey = new HashMap<>();
+
+        for (DSMap.Entry en : keyMap) {
+            String key = en.getKey().trim();
+            DSMap valueMaP = (DSMap)en.getValue();
+            // Only S, N and B are allowed as primary keys
+            if(valueMaP.firstEntry().getKey().equals("S")){
+                startKey.put(key, new AttributeValue().withS(valueMaP.firstEntry().getValue().toString()));
+            } else if(valueMaP.firstEntry().getKey().equals("N")){
+                startKey.put(key, new AttributeValue().withN(valueMaP.firstEntry().getValue().toString()));
+            } else if(valueMaP.firstEntry().getKey().equals("B")){
+                ByteBuffer buffer = ByteBuffer.wrap((valueMaP.firstEntry().getValue().toString().getBytes()));
+                startKey.put(key, new AttributeValue().withB(buffer));
+            }
+        }
+        return startKey;
+    }
+
+    private static PrimaryKey getPrimaryKey(DSMap keyMap) {
+        PrimaryKey primaryKey = new PrimaryKey();
+        for (DSMap.Entry en : keyMap) {
+            String key = en.getKey().trim();
+            DSMap valueMaP = (DSMap)en.getValue();
+            // Only S, N are allowed as primary keys
+            if(valueMaP.firstEntry().getKey().equals("S")){
+                primaryKey.addComponent(key, valueMaP.firstEntry().getValue().toString());
+            } else if(valueMaP.firstEntry().getKey().equals("N")){
+                primaryKey.addComponent(key, Long.parseLong(valueMaP.firstEntry().getValue().toString()));
+            }
+        }
+        return primaryKey;
+    }
+
+    public static Set getBibarySet(DSList list){
+        Set setC = new LinkedHashSet();
+        for (DSElement en : list) {
+            setC.add(en.toString().getBytes());
+        }
+        return setC;
+    }
+
+    public static Set getStringSet(DSList list){
+        Set setS = new LinkedHashSet();
+        for (DSElement en : list) {
+            setS.add(en.toString());
+        }
+        return setS;
+    }
+
+    public static Set getNumberSet(DSList list){
+        Set setN = new LinkedHashSet();
+        for (DSElement en : list) {
+            setN.add(Double.valueOf(en.toString()));
+        }
+        return setN;
+    }
+
+    public static Map parseHashMap(DSMap m){
```

Comment:
Map, List, Set, etc. should be parameterized (e.g. Map<String, String>)

Reply:
Done!

Reply:
I'm still seeing unparameterized references to ArrayList, List, and Set in Util.java, please change those as well.

Reply:
Done!